[Enhancement] Remove unused CG_ERR_MISSING_COMMAND error code

## Description

Shellcheck reports that `CG_ERR_MISSING_COMMAND` is unused in `config/command_guard.sh`.

## Current State
- File: `config/command_guard.sh` line 10
- Variable: `readonly CG_ERR_MISSING_COMMAND=1`
- Issue: Variable is defined but never referenced in code execution

## Analysis
The error code was likely intended for cases where commands are missing, but the current implementation uses `CG_ERR_NOT_FOUND` instead. The unused constant should be removed to clean up the code.

## References Found
- **Code**: `config/command_guard.sh` (definition and comment)
- **Documentation**: `docs/libraries/command_guard.rst` (mentions in error descriptions)
- **Tests**: `test/test-command_guard.bats` (exports the variable)

## Acceptance Criteria
- [ ] Remove the unused `CG_ERR_MISSING_COMMAND=1` line from `config/command_guard.sh`
- [ ] Remove `CG_ERR_MISSING_COMMAND` from the error list comment in `config/command_guard.sh`
- [ ] Update documentation in `docs/libraries/command_guard.rst` to remove references
- [ ] Update test file `test/test-command_guard.bats` to remove the export
- [ ] Run shellcheck to confirm the warning is resolved
- [ ] Run tests to ensure no regressions

## Files to Modify
- `config/command_guard.sh`
- `docs/libraries/command_guard.rst`
- `test/test-command_guard.bats`

## Labels
code-quality, shellcheck, cleanup

## Implementation Plan — Branch Created

Branch: `refactor/issue-24-remove-unused-cg-err-missing-command` (pushed, tracks origin)

### Detailed task breakdown

| # | File | Change |
|---|---|---|
| 1 | `config/command_guard.sh` L10 | Delete `readonly CG_ERR_MISSING_COMMAND=1` |
| 2 | `config/command_guard.sh` L55 | Remove `CG_ERR_MISSING_COMMAND,` from `# Errors:` comment |
| 3 | `test/test-command_guard.bats` L16 | Remove `CG_ERR_MISSING_COMMAND` from the `export` line |

### Testing strategy

- `shellcheck config/command_guard.sh` — confirms the SC2034 (unused variable) warning is gone
- `bats test/test-command_guard.bats` — full suite must pass with no regressions

### Documentation

No RST change needed — `docs/libraries/command_guard.rst` has no reference to this constant.

Proceeding to implementation.

--- a/config/command_guard.sh
+++ b/config/command_guard.sh
@@ -52,7 +52,7 @@ _cg_resolve_command_path() {
 #   guard -q  # no warning
 #   guard -- uname -login  # Treats "-login" as a command name
 # Errors:
-#   CG_ERR_MISSING_COMMAND, CG_ERR_INVALID_NAME, CG_ERR_NOT_FOUND
+#   CG_ERR_INVALID_NAME, CG_ERR_NOT_FOUND
 # Notes:
 #   Uses a restricted PATH ("/usr/bin:/bin") in a subshell to resolve the command.
 #   This avoids resolving through user-controlled PATH entries.

--- a/config/command_guard.sh
+++ b/config/command_guard.sh
@@ -7,7 +7,6 @@
 [[ -z ${__COMMAND_GUARD_SH_INCLUDED:-} ]] && __COMMAND_GUARD_SH_INCLUDED=1 || return 0
 
 # --- Public error codes --------------------------------------------------------
-readonly CG_ERR_MISSING_COMMAND=1
 readonly CG_ERR_INVALID_NAME=2
 readonly CG_ERR_NOT_FOUND=3
 

--- a/test/test-command_guard.bats
+++ b/test/test-command_guard.bats
@@ -13,7 +13,7 @@ setup_file() {
   # shellcheck source=config/command_guard.sh
   source "$LIB"
   export -f guard _cg_resolve_command_path
-  export CG_ERR_MISSING_COMMAND CG_ERR_INVALID_NAME CG_ERR_NOT_FOUND
+  export CG_ERR_INVALID_NAME CG_ERR_NOT_FOUND
 }
 
 # bats test_tags=guard,issue-24

--- a/test/test-command_guard.bats
+++ b/test/test-command_guard.bats
@@ -16,6 +16,16 @@ setup_file() {
   export CG_ERR_MISSING_COMMAND CG_ERR_INVALID_NAME CG_ERR_NOT_FOUND
 }
 
+# bats test_tags=guard,issue-24
+# Expected to fail until CG_ERR_MISSING_COMMAND is removed from command_guard.sh
+@test "issue-24: CG_ERR_MISSING_COMMAND is not defined after sourcing" {
+  # shellcheck disable=SC2016
+  run -0 bash --noprofile --norc -c '
+    source "$LIB"
+    [[ -z "${CG_ERR_MISSING_COMMAND+x}" ]]
+  '
+}
+
 # bats test_tags=guard
 @test "guard defines a function that shadows the command" {
   # shellcheck disable=SC2016

